samples/emoji-search/presenter/src/commonTest/kotlin/com/example/redwood/emojisearch/presenter/EmojiSearchWidgets.kt

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (C) 2023 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.redwood.emojisearch.presenter
+
+import app.cash.redwood.LayoutModifier
+import app.cash.redwood.RedwoodCodegenApi
+import app.cash.redwood.layout.widget.Column
+import app.cash.redwood.layout.widget.RedwoodLayoutWidgetFactory
+import app.cash.redwood.layout.widget.Row
+import app.cash.redwood.treehouse.lazylayout.widget.LazyColumn
+import app.cash.redwood.treehouse.lazylayout.widget.RedwoodTreehouseLazyLayoutWidgetFactory
+import com.example.redwood.emojisearch.widget.EmojiSearchWidgetFactories
+import com.example.redwood.emojisearch.widget.EmojiSearchWidgetFactory
+import com.example.redwood.emojisearch.widget.Image
+import com.example.redwood.emojisearch.widget.Text
+import com.example.redwood.emojisearch.widget.TextInput
+import example.values.TextFieldState
+
+// TODO(swankjesse): GENERATE THIS.
+
+@OptIn(RedwoodCodegenApi::class)
+fun EmojiSearchTester(): RedwoodTester {
+  return RedwoodTester(
+    EmojiSearchWidgetFactories(
+      EmojiSearch = MutableEmojiSearchWidgetFactory,
+      RedwoodLayout = MutableRedwoodLayoutWidgetFactory,
+      RedwoodTreehouseLazyLayout = MutableRedwoodTreehouseLazyLayoutWidgetFactory,
+    ),
+  )
+}
+
+@RedwoodCodegenApi
+private object MutableEmojiSearchWidgetFactory : EmojiSearchWidgetFactory<MutableWidget> {
+  override fun TextInput(): TextInput<MutableWidget> = MutableTextInput()
+
+  override fun Text(): Text<MutableWidget> = MutableText()
+
+  override fun Image(): Image<MutableWidget> = MutableImage()
+}
+
+@RedwoodCodegenApi
+private class MutableTextInput : TextInput<MutableWidget>, MutableWidget {
+  override val value: MutableWidget
+    get() = this
+
+  override var layoutModifiers: LayoutModifier = LayoutModifier
+
+  var state: TextFieldState? = null
+  var hint: String? = null
+  var onChange: ((TextFieldState) -> Unit)? = null
+
+  override fun state(state: TextFieldState) {
+    this.state = state
+  }
+
+  override fun hint(hint: String) {
+    this.hint = hint
+  }
+
+  override fun onChange(onChange: ((TextFieldState) -> Unit)?) {
+    this.onChange = onChange
+  }
+
+  override fun snapshot() = TextInputValue(layoutModifiers, state, hint, onChange)
```

Comment:
I could cache snapshot objects. This might need a way for children to know their parent widgets so they could invalidate 'em. Unclear whether this juice is worth the squeeze.

Reply:
Yeah I'm not sure that offers anything we need. This is exactly what the old code did. Plus it's all test-specific.